File: bioskopina_mobile/lib/models/bioskopina_watchlist.dart
import 'package:json_annotation/json_annotation.dart';

import 'bioskopina.dart';  // Import the Bioskopina model
part 'bioskopina_watchlist.g.dart';

@JsonSerializable()
class BioskopinaWatchlist {
  int? id;
  int? movieId;
  int? watchlistId;
  String? watchStatus;
  int? progress;
  DateTime? dateStarted;
  DateTime? dateFinished;
  Bioskopina? bioskopina;

  BioskopinaWatchlist(
      this.id,
      this.movieId,
      this.watchlistId,
      this.watchStatus,
      this.progress,
      this.dateStarted,
      this.dateFinished,
      this.bioskopina,
      );

  factory BioskopinaWatchlist.fromJson(Map<String, dynamic> json) =>
      _$BioskopinaWatchlistFromJson(json);

  Map<String, dynamic> toJson() => _$BioskopinaWatchlistToJson(this);
}

File: backend/Bioskopina.Model/BioskopinaWatchlist.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bioskopina.Model
{
    public class BWatchlist
    {
        public int Id { get; set; }

        public int MovieId { get; set; }

        public int WatchlistId { get; set; }

        public string WatchStatus { get; set; } = null!;

        public int Progress { get; set; }

        public DateTime? DateStarted { get; set; }

        public DateTime? DateFinished { get; set; }

        public virtual Bioskopina Bioskopina { get; set; } = null!;

        public virtual Watchlist Watchlist { get; set; } = null!;
    }
}
